src/xdsl/dialects/builtin.py

```diff
@@ -145,75 +144,75 @@ def from_index_int_value(value: int) -> IntegerAttr:
 
     @staticmethod
     @builder
-    def from_params(value: Union[int, IntAttr],
-                    typ: Union[int, Attribute]) -> IntegerAttr:
+    def from_params(value: int | IntAttr, typ: int | Attribute) -> IntegerAttr:
         value = IntAttr.build(value)
         if not isinstance(typ, IndexType):
             typ = IntegerType.build(typ)
         return IntegerAttr([value, typ])
 
 
 @irdl_attr_definition
-class ArrayAttr(Data[List[Attribute]]):
+class ArrayAttr(Data[List[A]]):
```

Comment:
Here, you are using the `A` defined in ir.
Could we instead use a new typevar for each generic class? Otherwise, a change in `ir` will have an effect here, and we probably won't think of updating this

Reply:
Type variables in Python are sufficiently weird.
The blank import of `ir` imports `A` as well that's why it is in scope.
Maybe we can have all type variables start with an underscore, so that they are not automatically imported everywhere?